src/pages/filter/D3Plot.svelte

```diff
@@ -0,0 +1,295 @@
+<!--
+  (c) 2023, Center for Computational Thinking and Design at Aarhus University and contributors
+ 
+  SPDX-License-Identifier: MIT
+ -->
+
+ <script lang="ts">
+    import { onMount, onDestroy } from "svelte";
+    import { get } from 'svelte/store';
+    import * as d3 from "d3";    
+    import {
+        FilterType,
+        determineFilter,
+    } from '../../script/datafunctions';
+    import { GestureData, gestures } from '../../script/stores/mlStore';
+    import { state } from "../../script/stores/uiStore";
+    import { getPrevData } from '../../script/stores/mlStore';
+
+
+    export let filter: FilterType;
+    export let fullScreen: boolean = false;
+
+    $: showLive = $state.isInputConnected;
+
+    const uniqueLiveDataID = 983095438740;
+
+    const plotSize = fullScreen ? 800 : 400;
+
+    const createLiveData = () => {
+      const liveData = getPrevData();
+      if (liveData === undefined) return undefined;
+      const filteredData: RecordingRepresentation = {
+        ID: uniqueLiveDataID,
+        gestureClass: "live",
+        x: filterFunction(liveData.x),
+        y: filterFunction(liveData.y),
+        z: filterFunction(liveData.z),
+      };
+      return filteredData;
+    };
+    
+    let plot: any = undefined;
+
+    let notMountedYet = true;
+
+    let publicClassList: string[] = [];
+
+    type RecordingRepresentation = {
+      ID: number;
+      gestureClass: string;
+      x: number;
+      y: number;
+      z: number;
+    }
+
+    type axis = "x" | "y" | "z";
+
+    const filterStrategy = determineFilter(filter);
+    const filterFunction = (data: number[]) => filterStrategy.computeOutput(data);
+    let color: d3.ScaleOrdinal<string, unknown> | undefined = undefined;
```

Comment:
It makes good sense, but the range() function is typed ScaleOrdinal<string, unknown, never>, so the compiler will not accept it.

Reply:
It depends on which overload you look at. Since we defined the range type, the type will actually be 
```ScaleOrdinal<string, string>;```
Because we provide the `<string>` generic type specification

See the documentation from d3
```ts
/**
 * Constructs a new ordinal scale with the specified range.
 * The domain defaults to the empty array.
 * If range is not specified, it defaults to the empty array; an ordinal scale always returns undefined until a non-empty range is defined.
 *
 * The generic corresponds to the data type of range elements.
 *
 * @param range An optional array of range values to initialize the scale with.
 */
export function scaleOrdinal<Range>(range?: Iterable<Range>): ScaleOrdinal<string, Range>;
```

Reply:
I know the visual studio inspector may not be helpful, but in my test, if you make this change it will stop complaining :)

Reply:
It works now. Thanks!